platform-tests/src/platform/acceptance/uaa_styling_test.go

```diff
@@ -0,0 +1,96 @@
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/PuerkitoBio/goquery"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/cf"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+	"io/ioutil"
+	"net/url"
+	"strings"
+)
+
+var _ = Describe("UAA authorization webpage styling", func() {
+	var authURL *url.URL
+	var authLoginURL url.URL
+	var authStylesheetURL url.URL
+	var customAppURL url.URL
+	var customLogoURL url.URL
+	var customStylesheetURL url.URL
+
+	BeforeEach(func() {
+		infoCommand := cf.Cf("curl", "/v2/info")
+		Expect(infoCommand.Wait(testConfig.DefaultTimeoutDuration())).To(Exit(0))
+
+		var infoResp struct {
+			AuthorizationEndpoint string `json:"authorization_endpoint"`
+		}
+		err := json.Unmarshal(infoCommand.Buffer().Contents(), &infoResp)
+		Expect(err).NotTo(HaveOccurred())
+
+		authURL, err = url.Parse(infoResp.AuthorizationEndpoint)
+		Expect(err).NotTo(HaveOccurred())
+
+		authLoginURL = *authURL
+		authLoginURL.Path = "/login"
+
+		authStylesheetURL = *authURL
+		authStylesheetURL.Path = "/resources/oss/stylesheets/application.css"
+
+		customAppURL = url.URL{
+			Scheme: "https",
+			Host:   "paas-uaa-assets." + testConfig.AppsDomain,
+		}
+		customLogoURL = customAppURL
+		customLogoURL.Path = "/images/product-logo.png"
+		customStylesheetURL = customAppURL
+		customStylesheetURL.Path = "/stylesheets/application.css"
+	})
+
+	Context("the login page", func() {
+		var authLoginDoc *goquery.Document
+
+		BeforeEach(func() {
+			response, err := httpClient.Get(authLoginURL.String())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(200))
+			authLoginDoc, err = goquery.NewDocumentFromResponse(response)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should be using our custom stylesheet", func() {
+			stylesheetHrefs := authLoginDoc.Find("link[rel=stylesheet]").Map(func(_ int, linkTag *goquery.Selection) string {
+				href, _ := linkTag.Attr("href")
+				return href
+			})
+			Expect(stylesheetHrefs).To(ConsistOf([]string{
+				"/vendor/font-awesome/css/font-awesome.min.css",
+				customStylesheetURL.String(),
+			}))
+		})
+
+		It("should be using our custom logo", func() {
+			styleCSS, err := authLoginDoc.Find("style").Html()
+			Expect(err).NotTo(HaveOccurred())
+			styleCSS = strings.TrimSpace(styleCSS)
+			expectedCSS := fmt.Sprintf(".header-image {background-image: url(%s);}", customLogoURL.String())
+			Expect(styleCSS).To(Equal(expectedCSS))
+		})
+	})
+
+	Context("the default stylesheet", func() {
+		It("should not have changed from our saved version", func() {
+			response, err := httpClient.Get(authStylesheetURL.String())
+			Expect(err).NotTo(HaveOccurred())
+			returnedBody, err := ioutil.ReadAll(response.Body)
+			Expect(err).NotTo(HaveOccurred())
+
+			expectedBody, err := ioutil.ReadFile("data/expected_uaa_default_stylesheet.css")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(returnedBody)).To(Equal(string(expectedBody)))
```

Comment:
We talked about adding some test on the assertion to describe what somebody should do if this test fails.